astropy/nddata/nddata.py

```diff
@@ -307,10 +307,38 @@ def __str__(self):
 
     def __repr__(self):
         prefix = self.__class__.__name__ + "("
-        data = np.array2string(self.data, separator=", ", prefix=prefix)
-        unit = f", unit='{self.unit}'" if self.unit is not None else ""
-
-        return f"{prefix}{data}{unit})"
+        if (
+            isinstance(self.data, (int, float, np.ndarray))
+            or np.issubdtype(float, self.data)
+            or np.issubdtype(int, self.data)
+        ):
+            # if data is an ndarray, get build a repr via Masked:
+            ma = Masked(self.data, mask=self.mask)
+            data_repr = repr(ma)
+            # strip the class name generated by Masked, and the final ")",
+            # and re-pad multi-line array outputs
+            after_first_paren = data_repr.index("(") + 1
+            old_prefix_spaces = " " * after_first_paren
+            new_prefix_spaces = " " * len(prefix)
+            data_repr = data_repr[after_first_paren:-1].replace(
+                old_prefix_spaces, new_prefix_spaces
+            )
+            unit = f", unit='{self.unit}'" if self.unit is not None else ""
+            return f"{prefix}{data_repr}{unit})"
+        else:
```

Comment:
Codecov says this "else" is not covered.

Reply:
@pllim This `else` covers cases where `data` is a dask array, see discussion in https://github.com/astropy/astropy/pull/14438#discussion_r1117116482. I'm not sure if that can/should be covered in our tests? cc @Cadair

Reply:
We do have dask here:

https://github.com/astropy/astropy/blob/7bc600506dad3c1ce697a68fcbbc395256a438a4/setup.cfg#L78

Reply:
Dask test coverage introduced 👌🏻